File: BI-TJV/semestral/rest-server/src/main/java/wrzecond/service/TjvAllergenService.java
package wrzecond.service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import wrzecond.allergen.TjvAllergenCreateDTO;
import wrzecond.allergen.TjvAllergenReadDTO;
import wrzecond.entity.TjvAllergen;
import wrzecond.repository.TjvAllergenRepository;

@Service
public class TjvAllergenService extends TjvService<TjvAllergen, TjvAllergenReadDTO, TjvAllergenCreateDTO> {

    @Autowired
    public TjvAllergenService (TjvAllergenRepository allergenRepository) {
        super (allergenRepository);
    }

    @Override
    protected TjvAllergenReadDTO toDTO (TjvAllergen entity) {
        return new TjvAllergenReadDTO(entity.getId(), entity.getName());
    }

    @Override
    protected TjvAllergen fromDTO (TjvAllergenCreateDTO dto) {
        return new TjvAllergen(dto.getName());
    }

    @Override
    protected TjvAllergen merge (TjvAllergen entity, TjvAllergenCreateDTO dto) {
        if (dto.getName() != null)
            entity.setName(dto.getName());
        return entity;
    }

}

File: BI-KOT/semestral/rest-server/src/main/kotlin/wrzecond/service/TjvAllergenService.kt
package wrzecond.service

import org.springframework.stereotype.Service
import wrzecond.TjvAllergenCreateDTO
import wrzecond.TjvAllergenReadDTO
import wrzecond.TjvAllergenUpdateDTO
import wrzecond.entity.TjvAllergen
import wrzecond.repository.TjvAllergenRepository

@Service
class TjvAllergenService (override val repository: TjvAllergenRepository)
: TjvService<TjvAllergen, TjvAllergenReadDTO, TjvAllergenCreateDTO, TjvAllergenUpdateDTO> (repository) {

    override fun TjvAllergen.toDTO() = TjvAllergenReadDTO(id, name)
    override fun TjvAllergenCreateDTO.toEntity() = TjvAllergen(name)
    override fun TjvAllergen.merge (dto: TjvAllergenUpdateDTO) = TjvAllergen(dto.name ?: name, id)

}